src/014_syntax_description.md

```diff
@@ -13,40 +13,48 @@ All security protocol theory are named and delimited by `begin` and `end`.
 We explain the non-terminals of the body in the following paragraphs.
 
     security_protocol_theory := 'theory' ident 'begin' body 'end'
-    body := (signature_spec | rule | restriction | lemma | formal_comment)+
+    body := (signature_spec | rule | restriction | lemma | formal_comment)*
 
 Here, we use the term signature more liberally to denote both the defined
 function symbols and the equalities describing their interaction.  Note that
 our parser is stateful and remembers what functions have been defined. It will
 only parse function applications of defined functions.
 
     signature_spec := functions | equations | built_in
-    functions      := 'functions' ':' (ident '/' arity) list
-    equations      := 'equations' ':' (term '=' term) list
+    functions      := 'functions' ':' function_sym (',' function_sym)*
+    function_sym   := ident '/' arity ['[private]']
     arity          := digit+
+    equations      := 'equations' ':' equation (',' equation)*
+    equation       := (term '=' term)
 
-Note that the equations must be subterm-convergent. Tamarin provides built-in
+Note that the equations must be subterm-convergent, and do not allow the use
```

Comment:
This is outdated, Tamarin now accepts any convergent set of equations that has the finite variant property.

Reply:
Addressed in my commit below.